runtime/CSharp/tests/issue-3510/ChannelTest.g4

```diff
@@ -0,0 +1,7 @@
+grammar ChannelTest;
+
+prog : INT+ EOF;
+INT : [0-9]+;
+AAA : [a-z]+ -> channel(HIDDEN); // Note--lowercase "channel"
+BBB : [A-Z]+ -> Channel(HIDDEN); // Note--uppercase "Channel"
+WS : [ \t\r\n] -> skip;
```

Comment:
It should be written using a descriptor (https://github.com/antlr/antlr4/tree/dev/runtime-testsuite/resources/org/antlr/v4/test/runtime/descriptors/LexerExec) to cover all runtimes but not C# only.

Also, please add tests on other commands if you can (see [lexer-commands](https://github.com/antlr/antlr4/blob/dev/doc/lexer-rules.md#lexer-commands) description):

* skip
* more
* popMode
* pushMode
* type
* channel

Reply:
OK, was wondering what those were for.

Reply:
BTW, we have similar tests but in `tool-testsuite`: https://github.com/antlr/antlr4/blob/dev/tool-testsuite/test/org/antlr/v4/test/tool/TestLexerActions.java But actually they should be moved to `runtime-testsuite`.